### PR TITLE
ref(data): implementing versions for real

### PR DESCRIPTION
**Pull request by arschles**
This PR does the following:

- Implements a working version of the `(github.com/deis/workflow-manager/data/availableVersionsFromAPI).Refresh` function that actually makes requests to the versions server.
- Moves the code in the `github.com/deis/workflow-manager/components` package into the `github.com/deis/workflow-manager/data` package

This work was previously completed in #37 as part of a bigger refactor. That refactor has since been split into multiple PRs, culminating with this one.

Depends on #49
Closes #37

### CODE REVIEW

File: data/data_test.go [hunk 1]
```diff
@@ -108,6 +148,101 @@ func TestGetAvailableVersions(t *testing.T) {
...
+	mockCluster := getMockCluster(t)
+	cluster, err := GetCluster(mocks.InstalledMockData{}, &mocks.ClusterIDMockData{}, mocks.LatestMockData{})
+	assert.NoErr(t, err)
+	assert.Equal(t, cluster.ID, mockCluster.ID, "ID value")
+	for i, component := range cluster.Components {
+		// verify that GetCluster returned the expected mock Component data
+		assert.Equal(t, component.Component, mockCluster.Components[i].Component, "Component value")
+		// verify that GetCluster returned the expected mock Version data
+		assert.Equal(t, component.Version, mockCluster.Components[i].Version, "Version property")
+		// TODO
```
Comment (vdice) [inline data/data_test.go]: Does this `TODO` still need to be finished for the PR?  If not, is there a ticket tracking it yet?

File: data/data_test.go [hunk 2]
```diff
@@ -108,6 +148,101 @@ func TestGetAvailableVersions(t *testing.T) {
...
+		assertUpdateAvailableAdded(latestComponent, cluster.Components[i], t)*/
+	}
+}
+
+func TestAddUpdateData(t *testing.T) {
+	mockCluster := getMockCluster(t)
+	// AddUpdateData should add an "UpdateAvailable" field to any components whose versions are out-of-date
+	err := AddUpdateData(&mockCluster, mocks.LatestMockData{})
+	assert.NoErr(t, err)
+	//TODO when newestVersion is implemented, actually test for the addition of "UpdateAvailable" fields
```
Comment (vdice) [inline data/data_test.go]: Might suggest adding the github ticket that tracks the implementation (and tests for) newestVersion here...

File: data/data_test.go [hunk 3]
```diff
@@ -108,6 +148,101 @@ func TestGetAvailableVersions(t *testing.T) {
...
+func TestGetInstalled(t *testing.T) {
+	cluster, err := GetInstalled(mockInstalledComponents{})
+	assert.NoErr(t, err)
+	assert.Equal(t, cluster.Components[0].Component.Name, mockComponentName, "Name value")
+	assert.Equal(t, cluster.Components[0].Component.Description, mockComponentDescription, "Description value")
+	assert.Equal(t, cluster.Components[0].Version.Version, mockComponentVersion, "Version value")
+}
+
+func TestParseJSONCluster(t *testing.T) {
+	const id = "f91378a6-a815-4c20-9b0d-77b205cd3ba4"
```
Comment (vdice) [inline data/data_test.go]: More a curiosity question here -- does this test component differentiate from a `mockInstalledComponent` as used in prev tests?  (Perhaps it isn't 'installed'?)

**Comment by vdice** [comment]
Added some comments but none blocking for this PR; LGTM!

File: data/data_test.go (hunk 2 above)
Comment (arschles) [inline data/data_test.go]: @vdice see https://github.com/deis/workflow-manager/issues/52. I'll add that issue to the comment as well

Comment (arschles) [inline data/data_test.go]: added in f4410e2

File: data/data_test.go (hunk 3 above)
Comment (arschles) [inline data/data_test.go]: well, this test is simply for ensuring that `ParseJSONCluster` can decode the expected JSON. but, the id here is a duplicate of the above, so I'll remove

Comment (arschles) [inline data/data_test.go]: fixed in https://github.com/deis/workflow-manager/pull/42/commits/32f8a0add8d19362050945add3f78403b833d34d

File: data/data_test.go (hunk 1 above)
Comment (arschles) [inline data/data_test.go]: added link to issue in https://github.com/deis/workflow-manager/pull/42/commits/edd945eb61ea39d16772e6c9f565e5486925827b

**Pull request merged**